src/main/java/gregtech/integration/jei/multiblock/MultiblockInfoRecipeWrapper.java

```diff
@@ -162,6 +175,13 @@ private void switchRenderPage(int amount) {
         }
     }
 
+    private void preparePlaceForParts(int recipeHeight) {
+        IGuiItemStackGroup itemStackGroup = recipeLayout.getItemStacks();
+
+        for (int i = 0; i < MAX_PARTS; ++i)
+            itemStackGroup.init(i, true, 18 * i - 180 * (i / 10), recipeHeight - PARTS_HEIGHT + 18 * (i / 10));
```

Comment:
There's quite a few magic numbers here. Maybe declare some constants so the reader can understand what is being computed here and why.

Reply:
There's really only 2 magic numbers missing, one is a standard, but I would suggest something like:

````
private static final int MAX_PARTS = 20;
+private static final int NO_PARTS_ROWS = 2; // 2 rows of parts
+private static final int PARTS_PER_ROW = MAX_PARTS / NO_PARTS_ROWS; // This is the 10
+private static final int SLOT_DIMENSIONS = 18; // This is the standard width/height of an item slot
-private static final int PARTS_HEIGHT = 36;
+private static final int PARTS_HEIGHT = SLOT_DIMENSIONS * NO_PARTS_ROWS; // This is the 36
+private static final int PARTS_WIDTH = SLOT_DIMENSIONS * PARTS_PER_ROW; // This is the 180
````

I am not sure it makes it anymore transparent that it is drawing a grid of item slots?
````
itemStackGroup.init(i, true, SLOT_DIMENSIONS * i - PARTS_WIDTH * (i / PARTS_PER_ROW), recipeHeight - PARTS_HEIGHT + SLOT_DIMENSIONS * (i / PARTS_PER_ROW));
````

You need to recognise the trick that (i / PARTS_PER_ROW) returns 0 for 0 to 9, 1 for 10 to 19, etc.

Maybe the following is clearer?
````
int row = i / PARTS_PER_ROW;
itemStackGroup.init(i, true, SLOT_DIMENSIONS * i - PARTS_WIDTH * row, recipeHeight - PARTS_HEIGHT + SLOT_DIMENSIONS * row);
````